File: dev/20190531/session/api/fizzbuzz/src/main/java/fizzbuzz/FizzBuzzList.java
package fizzbuzz;

import java.util.ArrayList;
import java.util.List;

public class FizzBuzzList implements IFizzBuzz {
  private final List<FizzBuzzValue> _list;

  public FizzBuzzList(List<FizzBuzzValue> list) {
    assert list.size() <= 100 : "FizzBuzzList can't generate over 100";
    _list = list;
  }

  @Override
  public String getValue() {
    return "";
  }

  public List<FizzBuzzValue> getList() {
    return _list;
  }

  public FizzBuzzList add(FizzBuzzValue value) {
    List<FizzBuzzValue> result = new ArrayList<>(_list);
    result.add(value);
    return new FizzBuzzList(result);
  }

}

File: dev/20190822/k2works/src/Domain/Model/FizzBuzzList.ts
import { FizzBuzzValue } from "./FizzBuzzValue";
import { IFizzBuzz } from "./IFizzBuzz";

export class FizzBuzzList implements IFizzBuzz {
  public get list(): FizzBuzzValue[] {
    return this._list;
  }
  // tslint:disable-next-line: variable-name
  private _list: FizzBuzzValue[];

  constructor(list: FizzBuzzValue[]) {
    if (list.length > 100) {
      throw new Error("FizzBuzzList can't generate over 100");
    }
    this._list = list;
  }

  public getValue(): string {
    return "";
  }

  public getList(): FizzBuzzValue[] {
    return this.list;
  }

  public add(value: FizzBuzzValue): FizzBuzzList {
    this._list.push(value);
    return new FizzBuzzList(this._list);
  }
}
